src/m365/spo/commands/group/group-user-add.ts

```diff
@@ -33,52 +34,95 @@ class SpoGroupUserAddCommand extends SpoCommand {
+        return this.getOnlyActiveUsers(args, logger)
+          .then((resolvedUsernameList: string[]): Promise<SharingResult> => {
+            if (this.verbose) {
+              logger.logToStderr(`Start adding Active user/s to SharePoint Group ${args.options.groupId ? args.options.groupId : args.options.groupName}`);
+            }
+
+            const data: any = {
+              url: args.options.webUrl,
+              peoplePickerInput: this.getFormattedUserList(resolvedUsernameList),
+              roleValue: `group:${groupId}`
+            };
+
+            const requestOptions: any = {
+              url: `${args.options.webUrl}/_api/SP.Web.ShareObject`,
+              headers: {
+                'Accept': 'application/json;odata=nometadata',
+                'Content-type': 'application/json;odata=verbose'
+              },
+              data: data,
+              responseType: 'json'
+            };
+
+            return request.post<SharingResult>(requestOptions);
+          });
       })
       .then((sharingResult: SharingResult): void => {
         if (sharingResult.ErrorMessage !== null) {
           return cb(new CommandError(sharingResult.ErrorMessage));
         }
+
         logger.log(sharingResult.UsersAddedToGroup);
 
         cb();
       }, (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
   }
 
+  private getGroupId(args: CommandArgs): Promise<number> {
+    if (args.options.groupId) {
+      return Promise.resolve(args.options.groupId);
+    }
+
+    const requestUrl: string = `${args.options.webUrl}/_api/web/sitegroups/GetByName('${encodeURIComponent(args.options.groupName as string)}')`;
+
+    const requestOptions: any = {
+      url: requestUrl,
+      headers: {
+        'accept': 'application/json;odata=nometadata'
+      },
+      responseType: 'json'
+    };
+
+    return request
+      .get<{ Id: number }>(requestOptions)
+      .then(response => {
+        const groupId: number | undefined = response.Id;
+
+        if (!groupId) {
+          return Promise.reject(`The specified group not exist in the SharePoint site`);
+        }
+
+        return Promise.resolve(groupId);
+      });
+  }
+
   private getOnlyActiveUsers(args: CommandArgs, logger: Logger): Promise<string[]> {
     if (this.verbose) {
       logger.logToStderr(`Removing Users which are not active from the original list`);
     }
 
     const activeUsernamelist: string[] = [];
-    return Promise.all(args.options.userName.split(",").map(singleUsername => {
+    const userInfo: string = args.options.userName ? args.options.userName : args.options.email!;
+
+    return Promise.all(userInfo.split(",").map(singleUsername => {
       const options: AadUserGetCommandOptions = {
-        userName: singleUsername.trim(),
         output: 'json',
         debug: args.options.debug,
         verbose: args.options.verbose
       };
+
+      if (args.options.userName) {
+        options['userName'] = singleUsername.trim();
```

Comment:
Let's not use indexers and instead use the named properties to avoid refactoring errors and typos.